src/typography/src/Label.js

```diff
@@ -1,14 +1,49 @@
 import React, { memo, forwardRef } from 'react'
-import Text from './Text'
+import cx from 'classnames'
+import PropTypes from 'prop-types'
+import Box from 'ui-box'
+import { useStyleConfig } from '../../hooks'
+
+const pseudoSelectors = {}
+const internalStyles = {}
 
 const Label = memo(
   forwardRef(function Label(props, ref) {
-    return <Text is="label" fontWeight={500} {...props} ref={ref} />
+    const {
+      className,
+      size = 400,
+      ...restProps
+    } = props
+
+    const { className: themedClassName, ...boxProps } = useStyleConfig(
+      'Heading',
```

Comment:
Doesn't this still couple `<heading />` and `<Label />`?

Reply:
I guess it does stylistically. I can decouple this more